Migrate to Gradle version catalog

vegaro opened the pull request:
Merging #1033 into main

vegaro commented on base-application.gradle:
```diff
@@ -1,7 +1,3 @@
-apply plugin: 'com.android.application'
-apply plugin: 'kotlin-android'
-apply plugin: 'kotlin-parcelize'
```
This wasn't being used

tonidero commented on build.gradle:
```diff
@@ -1,42 +1,27 @@
+    alias libs.plugins.androidx.navigation.safeargs apply false
+    //    TODO: is this needed?
+    alias libs.plugins.kotlin.kapt apply false
+    //    TODO: is this needed?
```
Hmm are these `TODO`'s  meant to be removed?

tonidero replied:
Looks like we only use parcelization in the `public` and `feature/amazon` modules, so I believe it should only be needed in those...

tonidero reviewed (commented):
Looks good but we should decide what to do with those todos

vegaro commented:
@tonidero removed those todos. I couldn't figure out how to remove parcelize from the root build.gradle, though

I also added a test bundle and extracted more versions

tonidero commented on common/build.gradle:
```diff
@@ -24,17 +31,13 @@ dependencies {
-    implementation "com.google.crypto.tink:tink-android:$tinkVersion"
-    implementation "com.android.billingclient:billing:$billingVersion"
+
+    implementation libs.kotlin.stdlib
```
We might not need to add the kotlin stdlib... But we can remove it in a different PR if needed.

tonidero replied:
Sorry I was confused, we talked and I think this is needed. In newer projects, the kotlin-dom is added instead, but we can keep this for now.

tonidero commented on build.gradle:
```diff
@@ -85,8 +68,6 @@ task detektAllBaseline(type: io.gitlab.arturbosch.detekt.DetektCreateBaselineTas
     exclude("**/test/**/*.kt")
 }
 
-apply plugin: 'org.jetbrains.dokka'
```
I see `kover` is not applied anywhere after this. Do we need to?

vegaro replied:
hmmm, it wasn't applied before. In `main` it gets applied in library.gradle (https://github.com/RevenueCat/purchases-android/pull/1059/files#diff-56729f0dbf9568b5dcd9f780979f8625a04144ffd5f9e0010e579d8d44570f84R43)

tonidero commented on build.gradle:
```diff
@@ -1,42 +1,25 @@
-    id "com.savvasdalkitsis.module-dependency-graph" version "0.9"
+    alias libs.plugins.mavenPublish apply false
+    alias libs.plugins.android.application apply false
+    alias libs.plugins.android.library apply false
```
So now we are adding both the application and library plugins here and applying them on the respective modules each. Is it necessary to specify both here?

vegaro replied:
Yes. Think this is the equivalent of adding the AGP plugin. Removing either plugin gives the same error as removing the parcelize (The request for this plugin could not be satisfied because the plugin is already on the classpath with an unknown version, so compatibility cannot be checked.). Removing `application` gives the issue in the apps, I am getting it in `api-tester`, and removing the library one gives it in `common`.

I think these plugins need to be applied both in the base and in the module's build.gradle. I guess it makes sense for the android plugins, because at the end of the day the need to build an android project, and this is the build.gradle of the project, but I am still not sure about the parcelize one.

tonidero commented on examples/MagicWeather/gradle/libs.versions.toml:
```diff
@@ -0,0 +1,27 @@
+agp = "8.0.0"
+androidxNavigation = "2.5.3"
+kotlin = "1.6.21"
+purchases = "6.1.1"
```
Should we update this to the latest 6.5.0?

tonidero commented on examples/MagicWeather/gradle/libs.versions.toml:
```diff
@@ -0,0 +1,27 @@
+[versions]
+agp = "8.0.0"
```
This should be 8.0.2

tonidero reviewed (approved):
Left a few more comments but it looks good!
